pkg/network/node/ovscontroller.go

```diff
@@ -71,11 +71,7 @@ func (oc *ovsController) AlreadySetUp(vxlanPort uint32) bool {
 }
 
 func (oc *ovsController) SetupOVS(clusterNetworkCIDR []string, serviceNetworkCIDR, localSubnetCIDR, localSubnetGateway string, mtu uint32, vxlanPort uint32) error {
-	err := oc.ovs.DeleteBridge(true)
```

Comment:
Why do we do `DeleteBridge` in SetupOVS? This causes all the vports in ovsdb to get deleted. and then when GetPodNetworkInfo is called, there is nothing in the db. are we expected to "re-add" the vports as the API server sync happens for already existing pod?

Reply:
`del-br br0` basically deletes the entire tree under the bridge.

Reply:
At this point we have already determined that something is wrong with `br0` and we can't use it as-is. We don't have code to synchronize it to a working state, so we have to destroy and recreate it. The ports get re-added later, in `reattachPods` in `pkg/network/node/node.go`

Reply:
aha, the problem is `reattachPods` relies on the `existingOFPodNetworks` to be there, the cache for which should have been built before the `DeleteBridge()` then. But right now we nuke the db when we delete the bridge, so the `existingOFPodNetworks` is basically an empty slice

```
 if networkChanged && len(existingOFPodNetworks) > 0 {
                err := node.reattachPods(existingPodSandboxes, existingOFPodNetworks)
                if err != nil {
                        return err
                }
 }
```